src/Files.App/Helpers/Win32/Win32Helper.Storage.cs

```diff
@@ -935,10 +943,72 @@ public static bool UnsetFileAttribute(string lpFileName, FileAttributes dwAttrs)
 			return Win32PInvoke.SetFileAttributesFromApp(lpFileName, lpFileInfo.dwFileAttributes & ~dwAttrs);
 		}
 
+		public static unsafe bool CanCompressContent(string path)
+		{
+			path = Path.GetPathRoot(path) ?? string.Empty;
+
+			// MAX_PATH = 260 + 1 (null termination)
+			var volumeName = new PWSTR();
+			var fileSystemName = new PWSTR();
+
+			uint serialNumber = 0;
+			uint maxComponentLength = 0;
+			uint fileSystemFlags = 0;
+
+			var success = PInvoke.GetVolumeInformation(
+				path,
+				volumeName,
+				261u,
+				&serialNumber,
+				&maxComponentLength,
+				&fileSystemFlags,
+				fileSystemName,
+				261u);
+
+			if (!success)
+				return false;
+
+			// 0x00000010 FILE_FILE_COMPRESSION
+			return (fileSystemFlags & 0x00000010) != 0;
```

Comment:
You can add FILE_FILE_COMPRESSION to CsWin32 and use from `PInvoke.FILE_FILE_COMPRESSION`

Reply:
What's the enum name?

Reply:
This is not a member of an enum but a **constant** value in C.
You can add FILE_FILE_COMPRESSION and use it like PInvoke.FILE_FILE_COMPRESSION

Reply:
This should be resolved with 31289160e0e62cb693cbe5d85fc6026355b58fc7